### ISSUE TITLE
Bad setuptools_scm version 0.0.0 when installed from a .tar.gz archive

### ISSUE THREAD

**Issue by bcroq** [pypa/setuptools-scm#209]
When setuptools_scm is installed from an sdist archive, it is installed as version 0.0.0.

E.g. when installing version 1.15.6:
```shell
root@79ddc1fccb0d:/# easy_install https://pypi.python.org/packages/03/6d/aafdd01edd227ee879b691455bf19895091872af7e48192bea1758c82032/setuptools_scm-1.15.6.tar.gz
Downloading https://pypi.python.org/packages/03/6d/aafdd01edd227ee879b691455bf19895091872af7e48192bea1758c82032/setuptools_scm-1.15.6.tar.gz
Processing setuptools_scm-1.15.6.tar.gz
Writing /tmp/easy_install-boEekB/setuptools_scm-1.15.6/setup.cfg
Running setuptools_scm-1.15.6/setup.py -q bdist_egg --dist-dir /tmp/easy_install-boEekB/setuptools_scm-1.15.6/egg-dist-tmp-j5W41h
/usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'use_scm_version'
  warnings.warn(msg)
Moving setuptools_scm-0.0.0-py2.7.egg to /usr/local/lib/python2.7/site-packages
Adding setuptools-scm 0.0.0 to easy-install.pth file

Installed /usr/local/lib/python2.7/site-packages/setuptools_scm-0.0.0-py2.7.egg
Processing dependencies for setuptools-scm==0.0.0
Finished processing dependencies for setuptools-scm==0.0.0

root@79ddc1fccb0d:/# pip freeze
setuptools-scm==0.0.0
```

**Comment by RonnyPfannschmidt** [comment]
@jaraco is there anything we can do to prevent this behaviour?

**Comment by bcroq** [comment]
This 0.0.0 version is a problem when installing packages that ask for a minimal version of setuptools_scm in their `setup_requires`, eg. pytest-runner:

```shell
root@b371b52ab48c:/# pip download --no-binary :all: -d /tmp pytest-runner setuptools-scm
Collecting pytest-runner
  Downloading pytest-runner-3.0.tar.gz
  Saved /tmp/pytest-runner-3.0.tar.gz
Collecting setuptools-scm
  Downloading setuptools_scm-1.15.6.tar.gz
  Saved /tmp/setuptools_scm-1.15.6.tar.gz
Successfully downloaded pytest-runner setuptools-scm
root@b371b52ab48c:/# easy_install -H None -f /tmp pytest-runner
Searching for pytest-runner
Best match: pytest-runner 3.0
Processing pytest-runner-3.0.tar.gz
Writing /tmp/easy_install-xn4Hlm/pytest-runner-3.0/setup.cfg
Running pytest-runner-3.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-xn4Hlm/pytest-runner-3.0/egg-dist-tmp-BVL_OD
/usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'use_scm_version'
  warnings.warn(msg)

Installed /tmp/easy_install-xn4Hlm/pytest-runner-3.0/.eggs/setuptools_scm-0.0.0-py2.7.egg
Traceback (most recent call last):
  File "/usr/local/bin/easy_install", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 2279, in main
    **kw
  File "/usr/local/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/local/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/local/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 411, in run
    self.easy_install(spec, not self.no_deps)
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 672, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 698, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 879, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1118, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/usr/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1104, in run_setup
    run_setup(setup_script, args)
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 257, in run_setup
    raise
  File "/usr/local/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 199, in setup_context
    yield
  File "/usr/local/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 170, in save_modules
    saved_exc.resume()
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 145, in resume
    six.reraise(type, exc, self._tb)
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 158, in save_modules
    yield saved
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 199, in setup_context
    yield
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 254, in run_setup
    _execfile(setup_script, ns)
  File "/usr/local/lib/python2.7/site-packages/setuptools/sandbox.py", line 49, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-xn4Hlm/pytest-runner-3.0/setup.py", line 68, in <module>
  File "/usr/local/lib/python2.7/distutils/core.py", line 111, in setup
    _setup_distribution = dist = klass(attrs)
  File "/usr/local/lib/python2.7/site-packages/setuptools/dist.py", line 335, in __init__
    self.fetch_build_eggs(attrs['setup_requires'])
  File "/usr/local/lib/python2.7/site-packages/setuptools/dist.py", line 456, in fetch_build_eggs
    replace_conflicting=True,
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 867, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'setuptools_scm>=1.15.0' distribution was not found and is required by the application

```

**Comment by RonnyPfannschmidt** [comment]
as a temporary workaround please ensure you have setuptools_scm  installed by pip in your environment

in general its very helpful to avoid easy_install since its sandboxing interferes with setuptools_scm being albe to self-install

**Comment by jaraco** [comment]
hgtools [had a specialized approach](https://github.com/jaraco/hgtools/blob/6.3/setup.py) to handle this condition.

```
$ mkdir -p foo/lib/python/site-packages
$ PYTHONUSERBASE=foo easy_install --user hgtools==6.3
Searching for hgtools==6.3
Reading https://pypi.python.org/simple/hgtools/
Downloading https://pypi.python.org/packages/9e/50/e461d5df23be4a87f451f1700c93f94b48bdf6d17afd54a11176871b8c7c/hgtools-6.3.zip#md5=584d74b81b1efae3604c53086d1a3acb
Best match: hgtools 6.3
Processing hgtools-6.3.zip
Writing /var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/easy_install-dmsu429v/hgtools-6.3/setup.cfg
Running hgtools-6.3/setup.py -q bdist_egg --dist-dir /var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/easy_install-dmsu429v/hgtools-6.3/egg-dist-tmp-5gqjolzu
zip_safe flag not set; analyzing archive contents...
Copying hgtools-6.3-py3.6.egg to /Users/jaraco/Dropbox/code/public/setuptools_scm/foo/lib/python/site-packages
Adding hgtools 6.3 to easy-install.pth file

Installed /Users/jaraco/Dropbox/code/public/setuptools_scm/foo/lib/python/site-packages/hgtools-6.3-py3.6.egg
Processing dependencies for hgtools==6.3
Finished processing dependencies for hgtools==6.3
```

**Comment by RonnyPfannschmidt** [comment]
i believe this has been fixed with #296 and is released in 3.0.6
